opta/nice_subprocess.py

```diff
@@ -60,65 +74,121 @@ def nice_run(  # type: ignore # nosec
     exit_timeout: Optional[float] = None,
     tee: bool = True,
     check: bool = False,
+    use_asyncio_nice_run=False,
     **kwargs,
 ) -> CompletedProcess:
 
-    try:
-        listargs = list(popenargs)
-        listargs[0].insert(0, "exec")
-        popenargs = tuple(listargs)
-        log_to_datadog(
-            "Calling subprocess with these arguments:\n" + " ".join(*popenargs), "INFO"
-        )
+    if (
+        use_asyncio_nice_run is False
```

Comment:
what about datadog logging? there is also a different behavior?
to JD's point, sometimes it's not desirable to log to datadog, the shell is a good example.
we could document that as well, or expose a specific boolean.
if asyncio && datadog logging are not supported together right now, we can also raise an exception so it would fail when testing it.

Reply:
In the old nice process there was no datadog logging.

There is also a switch in the nice_subprocess function to turn off the logging when needed, again which functions need to be quieter needs to be discussed and then the code put into place. This ticket is only the hotfix for the bug.

Reply:
go it, let's do hotfix and discuss later